src/scanoss/agent/main.c

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*!
+ * main.c
+ *
+ * The SCANOSS Agent for Fossology tool
+ *
+ * Copyright (C) 2018-2021 SCANOSS.COM
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "snippet_scan.h"
+#include <stdio.h>
+#ifdef COMMIT_HASH_S
+char BuildVersion[] = "scanoss build version: " VERSION_S " r(" COMMIT_HASH_S ").\n";
+#else
+char BuildVersion[] = "scanoss build version: NULL.\n";
+#endif
+char *baseTMP = "/tmp/scanoss";
+int Agent_pk;
+
+
+int createTables(PGconn* pgConn)
+{
+  char sql[8192];
+  PGresult* result;
+
+  if (!fo_tableExists(pgConn, "scanoss_fileinfo")) {
+   
+    snprintf(sql, sizeof(sql), "\
+          CREATE TABLE scanoss_fileinfo (\
+	          pfile_fk int4 NOT NULL,\
+            matchtype text NULL,\
+            lineranges text NULL,\
+            purl varchar NULL,\
+            url varchar NULL,\
+            filepath varchar NULL,\
+            fileinfo_pk serial4 NOT NULL\
+          );");
+     
+    result = PQexec(pgConn, sql);
+    if (fo_checkPQcommand(pgConn, result, sql, __FILE__, __LINE__)) {
+     
+    // Can 't create table scanoss_fileinfo
+  }
+  }
+}
+
+
+
+
+
+
+
+
+
+
+/*!
+ * \brief main function for the pkgagent
```

Comment:
Please change comment.

```suggestion
 * \brief main function for the scanoss
```